### PR TITLE
List Mask (MASK)

### PR DESCRIPTION
**Pull request by diablax**
- Official project URL: https://mask.team
- Official block explorer URL: http://explorer.mask.team

### CODE REVIEW

**Review by cbeams — CHANGES_REQUESTED** [review]
NACK. Squash to a single commit per instructions at https://bisq.network/list-asset, and use blank lines per convention in import statements for the `Mask` class.

**Comment by diablax** [comment]
squashed and imports fixed.

**Comment by cbeams** [comment]
From https://docs.bisq.network/exchange/howto/list-asset.html#step-5-create-a-well-formed-git-commit:

image

**Comment by diablax** [comment]
My apologies Chris, I have now added the changes.

File: assets/src/main/java/bisq/asset/coins/Mask.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset.coins;
+
+import bisq.asset.Coin;
+
```
Comment (cbeams) [inline assets/src/main/java/bisq/asset/coins/Mask.java]:
```suggestion
```

File: assets/src/test/java/bisq/asset/coins/MaskTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    public void testValidAddresses() {
+        assertValidAddress("MbxYjp38aUXBuESwsFv8YmRvbQvMhNyJygU6ViLCjM4sUNqFjsHQim9dvzp9p8BVTjdsRkVNrC1Zy3NJRb18hav3CPe5eWn");
+        assertValidAddress("MeGcanFnSr4bJFuNoHogCBdDCsqDrNu5njPc1Yh1DfsTUTL5dLbbtE119f4vztxXu6fFCKWRmpqjABdDyGrzMDkhTC38WwS");
+    }
+
+    @Test
+    public void testInvalidAddresses() {
+        assertInvalidAddress("");
+	assertInvalidAddress("Wecj8");
```
Comment (cbeams) [inline assets/src/test/java/bisq/asset/coins/MaskTest.java]:
```suggestion
        assertInvalidAddress("Wecj8");
```

File: assets/src/test/java/bisq/asset/coins/MaskTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	assertInvalidAddress("MsopefFnSr4bJFuNoHogCBdDCsqDrNu5Pc1Yh1DfsTUTL5dLbbtE119f4vztxXu6fFCKWRmpqjABdDyGrzMDkhTC38gWw");
```
Comment (cbeams) [inline assets/src/test/java/bisq/asset/coins/MaskTest.java]:
```suggestion
        assertInvalidAddress("MsopefFnSr4bJFuNoHogCBdDCsqDrNu5Pc1Yh1DfsTUTL5dLbbtE119f4vztxXu6fFCKWRmpqjABdDyGrzMDkhTC38gWw");
```

File: assets/src/test/java/bisq/asset/coins/MaskTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	assertInvalidAddress("MeGcanuyt4bJFuNoHogCBdDCsqDrNu5njPc1Yh1DfsTUTL5dLbbtE119f4vztxXu6fFCKWRmpqujABdDyGrzMDkhTC38WwSx");
```
Comment (cbeams) [inline assets/src/test/java/bisq/asset/coins/MaskTest.java]:
```suggestion
         assertInvalidAddress("MeGcanuyt4bJFuNoHogCBdDCsqDrNu5njPc1Yh1DfsTUTL5dLbbtE119f4vztxXu6fFCKWRmpqujABdDyGrzMDkhTC38WwSx");
```

**Review by cbeams — CHANGES_REQUESTED** [review]
NACK, per suggested formatting changes.

**Comment by diablax** [comment]
Again, apologies. Did not notice these messy lines on nano, perhaps they were formatted differently on my machine.

**Comment by diablax** [comment]
Let me know if there is anything else to do.

**Comment by cbeams** [comment]
From https://docs.bisq.network/exchange/howto/list-asset.html#step-6-submit-your-pull-request

image

The reason I need this is to clean up the body of your commit. You squashed it, but left all the comments from the squashed commits in the body of the remaining commit. The body should be empty. You can do that yourself and force push, or give me rights to do it. I'll merge afterward.

**Comment by diablax** [comment]
Feel free to amend. Edits are enabled.

**Comment by diablax** [comment]
Would it be easier to just submit a new PR?

**Comment by diablax** [comment]
I have cleaned up the code and will resubmit PR to comply with the rules.

**Comment by cbeams** [comment]
Please just reopen this PR with a force-pushed cleaned up version of your commit. There's no need to open a new PR.

**Comment by cbeams** [comment]
Wait. This PR now shows 2,085 files changed. Go ahead and resubmit unless you know how to fix that. Note that you submitted this PR from your `master` branch, which is something we ask you not to do in the instructions as well. This may be the source of the problem, especially if you pulled recent changes to upstream master.

**Comment by cbeams** [comment]
It's because you deleted your fork of the repository.

**Comment by diablax** [comment]
Yes I will resubmit everything properly and make sure it is on a different branch. Apologies for taking up so much of your time.

**Pull request closed without merging**